### PR TITLE
feat(FEC-10607): add current frozen version

### PR DESCRIPTION
**Pull request by RoyBregman**
Note: Needs ui/ux redesign

### CODE REVIEW

File: app/js/services/services.js [hunk 1]
```diff
@@ -585,6 +586,7 @@ KMCServices.factory('PlayerService', ['$http', '$modal', '$log', '$q', 'apiServi
 					'uiConf:description': data.description ? data.description : ''
 				};
 				request['uiConf:confVars'] = JSON.stringify({versions: playersService.getPlayerAndPluginsVersionObj(data), langs: data.playerLangCodes});
+				request['uiConf:html5Url'] = data.html5Url;
```
Comment (yairans) [inline app/js/services/services.js:589]: what html5Url holds for latest and beta?

File: app/js/services/services.js [hunk 2]
```diff
@@ -638,6 +640,34 @@ KMCServices.factory('PlayerService', ['$http', '$modal', '$log', '$q', 'apiServi
...
+							playersService.playerProductVersion = JSON.parse(kmc.vars.studioV3.playerConfVars);
+						} catch (e) {
+							console.error(e);
+						}
+					}
+					playersService.playerProductVersion = angular.isObject(playersService.playerProductVersion) &&  playersService.playerProductVersion.version ? playersService.playerProductVersion.version : "";
+				}
+				return playersService.playerProductVersion;
+			},
+            'getPlayerBetaProductVersion': function () {
```
Comment (yairans) [inline app/js/services/services.js:657]: I would expose one api getPlayerProductVersion which gets param 'latest' to 'beta'

File: app/templates/playerVersion.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<!--<div ng-class="{disabled: (plugin!=null && !plugin.enabled) || playerData.playerVersion=='beta'}" class="ng-scope">-->
```
Comment (yairans) [inline app/templates/playerVersion.html:1]: remove comments

**Review by yairans — CHANGES_REQUESTED** [review]

File: app/js/controllers/EditCtrl.js [hunk 4]
```diff
@@ -938,6 +996,16 @@ KMCMenu.controller('EditCtrl', ['$scope','$http', '$timeout','PlayerData','Playe
 			utilsSvc.alert('Save Player Settings','Some plugin features values are invalid. The player cannot be saved.');
 		}else{
 			var savePlayer = function () {
+				if ($scope.playerData.autoUpdate) {
+					$scope.playerData.html5Url = "";
+					$scope.playerData.frozenProductVersion = $scope.playerData.latestPlayerProductVersion;
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply app/js/controllers/EditCtrl.js:1001]: No, a bit different, maybe I will add comments to explain

File: app/js/services/services.js (hunk 1 above)
Comment (RoyBregman) [reply app/js/services/services.js:589]: Nothing - Empty string

File: app/templates/playerVersion.html (hunk 3 above)
Comment (RoyBregman) [reply app/templates/playerVersion.html:1]: done

File: app/js/controllers/EditCtrl.js [hunk 5]
```diff
@@ -1,3 +1,38 @@
+const UNKNOWN_PRODUCT_VERSION = "Unknown"
+const tagToProductVersion = {
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply app/js/controllers/EditCtrl.js]: done

File: app/js/controllers/EditCtrl.js [hunk 6]
```diff
@@ -27,6 +62,23 @@ KMCMenu.controller('EditCtrl', ['$scope','$http', '$timeout','PlayerData','Playe
...
+							$scope.playerData['frozenProductVersion'] = $scope.playerData.html5Url.version;
+						}
+					}
+					if(!$scope.playerData['frozenProductVersion']){
+						$scope.playerData['frozenProductVersion'] = tagToProductVersion[versionsObj[playerName]] || UNKNOWN_PRODUCT_VERSION ;
+					}
+				} catch (e) {
+					$scope.playerData['frozenProductVersion'] = tagToProductVersion[versionsObj[playerName]] || UNKNOWN_PRODUCT_VERSION;
+				}
+			}
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply app/js/controllers/EditCtrl.js]: done

File: app/js/services/services.js (hunk 2 above)
Comment (RoyBregman) [reply app/js/services/services.js]: done

File: app/js/controllers/EditCtrl.js [hunk 7]
```diff
@@ -27,6 +28,24 @@ KMCMenu.controller('EditCtrl', ['$scope','$http', '$timeout','PlayerData','Playe
...
+			PlayerService.getTagToProductVersion().success(function (tagToProductVersion) {
+				if (!autoUpdate){
+					try {
+						if ($scope.playerData.html5Url){
+							$scope.playerData.html5Url = JSON.parse($scope.playerData.html5Url);
+							if ($scope.playerData.html5Url.version) {
+								$scope.playerData['frozenProductVersion'] = $scope.playerData.html5Url.version;
+							}
+						}
+						$scope.playerData['frozenProductVersion'] = $scope.playerData['frozenProductVersion'] || tagToProductVersion[versionsObj[playerName]] || UNKNOWN_PRODUCT_VERSION;
```
Comment (yairans) [inline app/js/controllers/EditCtrl.js:40]: move this line after the try-catch and leave the catch empty

File: app/js/services/services.js [hunk 8]
```diff
@@ -217,6 +217,36 @@ KMCServices.factory('PlayerService', ['$http', '$modal', '$log', '$q', 'apiServi
...
+					} catch (e) {
+						console.error(e);
+					}
+				}
+				playersService.playerProductVersion = angular.isObject(playersService.playerProductVersion) &&  playersService.playerProductVersion.version ? playersService.playerProductVersion.version : "";
+			}
+			return playersService.playerProductVersion;
+		};
+
+		var getBetaPlayerProductVersion = function(){
```
Comment (yairans) [inline app/js/services/services.js:235]: Too much duplicated code. I meant to expose only one API and use the param. like `kmc.vars.studioV3.player${param}ConfVars`

**Review by yairans — CHANGES_REQUESTED** [review]

File: app/js/services/services.js (hunk 8 above)
Comment (RoyBregman) [reply app/js/services/services.js]: done, please see

File: app/js/controllers/EditCtrl.js (hunk 7 above)
Comment (RoyBregman) [reply app/js/controllers/EditCtrl.js]: done

**Review by yairans — APPROVED** [review]

**Pull request merged**